BedrockServer.cpp

```diff
@@ -852,18 +852,20 @@ void BedrockServer::worker(int threadId)
             // the `_futureCommitCommands` queue.
             if (state == SQLiteNode::FOLLOWING && command->escalateImmediately && !command->complete) {
                 if (_escalateOverHTTP) {
-                    if (_clusterMessenger.sendToLeader(*command)) {
-                        SINFO("Immediately escalating " << command->request.methodLine << " to leader.");
-                        waitForHTTPS(move(command));
+                    SINFO("Beginning immediately escalating " << command->request.methodLine << " to leader.");
+                    if (_clusterMessenger.runOnLeader(*command)) {
+                        // command->complete is now true for this command. It will get handled a few lines below.
+                        SINFO("Finished immediately escalating " << command->request.methodLine << " to leader.");
                     } else {
                         SWARN("Couldn't immediately escalate command " << command->request.methodLine << " to leader, queuing normally.");
                         _commandQueue.push(move(command));
+                        continue;
                     }
                 } else {
                     SINFO("Immediately escalating " << command->request.methodLine << " to leader. Sync thread has " << _syncNodeQueuedCommands.size() << " queued commands.");
                     _syncNodeQueuedCommands.push(move(command));
+                    continue;
```

Comment:
I'm not seeing the reason for moving the continue. If `_escalateOverHTTP` returns false, now, we'll exit the conditional and keep processing, which potentially changes the behavior a lot, right?

Reply:
If `_escalateOverHTTP` is false, we'll hit the `continue` that this comment is on, instead of the one that was removed below on line 866.

The difference is only when `_escalateOverHTTP` is `true`, where we don't `continue` if `runOnLeader` returns true, and there's a comment about that right there.

Reply:
Ah, yes, I misspoke `_escalateOverHTTP` is `true` is what I meant, sorry.

Reply:
When it's true this happens:
https://github.com/Expensify/Bedrock/pull/1222/files#diff-85aada0d0eb5de20bbc68017e21f8051525423072c9bfd097c7ca324d25c6125R857

Reply:
And this is where it gets handled a few lines below:
https://github.com/Expensify/Bedrock/blob/8ba56db5ed7c1f03b2335901029d9262af963e5b/BedrockServer.cpp#L889